server/portal/apps/auth/models.py

```diff
@@ -81,7 +82,11 @@ def client(self):
         with transaction.atomic():
             if self.expired:
                 try:
-                    client.refresh_tokens()
+                    lock = Lock()
```

Comment:
creating a lock object within this client function will not achieve mutual exclusion. A new lock object is created always and two threads are not looking at same lock anymore.

The `lock = Lock()` needs to be created at class level.